wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java

```diff
@@ -64,15 +68,34 @@ public Object getFieldValue(final Field field, final Object fieldOwner)
 				{
 					boolean optional = injectAnnotation != null && injectAnnotation.optional();
 					Annotation bindingAnnotation = findBindingAnnotation(field.getAnnotations());
-					final IProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+					final GuiceProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+
+					Object cachedValue = cache.get(locator);
```

Comment:
I'm going to revert all non-Objenesis caches.
This should be in separate ticket.